assignment2/assignment2.py

```
@@ -0,0 +1,183 @@
+#task2
+
+import csv
+import traceback
+
+def read_employees():
+    result = {}
+    rows = []
+    try:
+        with open("../csv/employees.csv", newline='') as f:
+            reader = csv.reader(f)
+            for i, row in enumerate(reader):
+                if i == 0:
+                    result["fields"] = row
+                else:
+                    rows.append(row)
+        result["rows"] = rows
+        return result
+
+    except Exception as e:
+        trace_back = traceback.extract_tb(e.__traceback__)
+        stack_trace = list()
+        for trace in trace_back:
+            stack_trace.append(
+                f'File : {trace[0]} , Line : {trace[1]}, Func.Name : {trace[2]}, Message : {trace[3]}'
+            )
+        print(f"Exception type: {type(e).__name__}")
+        message = str(e)
+        if message:
+            print(f"Exception message: {message}")
+        print(f"Stack trace: {stack_trace}")
+        exit(1)
```

This exit(1) works, but in general, it's better to avoid abruptly stopping your script unless absolutely necessary. You could raise the error again or return None, depending on how critical this failure is for the rest of your program.